src/vm/gccover.cpp

```diff
@@ -1745,23 +1837,34 @@ void DoGcStress (PCONTEXT regs, MethodDesc *pMD)
 
     CONSISTENCY_CHECK(!pThread->HasPendingGCStressInstructionUpdate());
 
-#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_) || defined(_TARGET_ARM_) || defined(_TARGET_ARM64_)
-    if (afterCallProtect) 
+    if (numberOfRegs != 0)
     {
-#ifdef _TARGET_AMD64_
-        regs->Rax = retVal;
+        if (afterCallProtect[0])
+        {
+#if defined(_TARGET_AMD64_)
+            regs->Rax = retValRegs[0]; // I should cut my own hands for writing this.
 #elif defined(_TARGET_X86_)
-        regs->Eax = retVal;
+            regs->Eax = retValRegs[0];
 #elif defined(_TARGET_ARM_)
-        regs->R0 = retVal;
+            regs->R0 = retValRegs[0];
 #elif defined(_TARGET_ARM64_)
-        regs->X[0] = retVal;
+            regs->X[0] = retValRegs[0];
 #else
-        PORTABILITY_ASSERT("DoGCStress - return register");
-#endif
+            PORTABILITY_ASSERT("DoGCStress - return register");
+#endif            
+        }
+
+        if (afterCallProtect[1])
+        {
+#if defined(_TARGET_AMD64_) && defined(PLATFORM_UNIX)
+            regs->Rdx = retValRegs[numberOfRegs - 1];
```

Comment:
Why is this not just `retValRegs[1]`?

Reply:
When we protect only the second we want to restore Rdx from retValRegs[0], and from retValRegs[1] when we protect both.